pkg/synccompactor/compactor.go

```diff
@@ -82,13 +84,49 @@ func (c *Compactor) Compact(ctx context.Context) (*CompactableSync, error) {
 	// Move last compacted file to the destination dir
 	finalPath := path.Join(c.destDir, fmt.Sprintf("compacted-%s.c1z", base.SyncID))
 	if err := os.Rename(base.FilePath, finalPath); err != nil {
-		return nil, fmt.Errorf("failed to move compacted file to destination: %w", err)
+		if strings.Contains(err.Error(), "cannot move the file to a different disk drive") {
```

Comment:
I wonder if the message is different based on the user's locale setting.

Here's some example code I noticed that detects the exact error on windows & unix and changes its behavior: https://github.com/cloudfoundry/bosh-utils/commit/0dc9b4fb252fe9d2a1fe45e269dab6cd353ab306

Maybe that's a little more robust.

Reply:
nice find